src/hyp3_opera_rtc/dem.py

```diff
@@ -1,22 +1,56 @@
-from concurrent.futures import ThreadPoolExecutor
-from itertools import product
+import os
+from collections.abc import Callable
 from pathlib import Path
-from tempfile import TemporaryDirectory
 
+import backoff
 import numpy as np
 import shapely
 import shapely.ops
 import shapely.wkt
-from hyp3lib.fetch import download_file
-from osgeo import gdal
-from shapely.geometry import LinearRing, Polygon
+from osgeo import gdal, osr
+from shapely.geometry import LinearRing, Polygon, box
 
 
 gdal.UseExceptions()
-URL = 'https://nisar.asf.earthdatacloud.nasa.gov/STATIC/DEM/v1.1/EPSG4326'
+
+EARTH_APPROX_CIRCUMFERENCE = 40075017.0
+EARTH_RADIUS = EARTH_APPROX_CIRCUMFERENCE / (2 * np.pi)
+
+
+def margin_km_to_deg(margin_in_km: float) -> float:
+    """Converts a margin value from kilometers to degrees."""
+    km_to_deg_at_equator = 1000.0 / (EARTH_APPROX_CIRCUMFERENCE / 360.0)
+    margin_in_deg = margin_in_km * km_to_deg_at_equator
+    return margin_in_deg
+
+
+def margin_km_to_longitude_deg(margin_in_km: float, lat: float) -> float:
+    """Converts a margin value from kilometers to degrees as a function of latitude."""
+    delta_lon = 180 * 1000 * margin_in_km / (np.pi * EARTH_RADIUS * np.cos(np.pi * lat / 180))
+    return delta_lon
+
+
+def polygon_from_bounds(bounds: tuple[float, float, float, float]) -> Polygon:
+    """Create a polygon (EPSG:4326) from the lat/lon coordinates corresponding to a provided bounding box."""
+    lon_min, lat_min, lon_max, lat_max = bounds
+    # note we can also use the center lat here
+    lat_worst_case = max([lat_min, lat_max])
+    margin_in_km = 50.0
+    lat_margin = margin_km_to_deg(margin_in_km)
+    lon_margin = margin_km_to_longitude_deg(margin_in_km, lat=lat_worst_case)
+    # Check if the bbox crosses the antimeridian and apply the margin accordingly
+    # so that any resultant DEM is split properly by check_dateline
+    if lon_max - lon_min > 180:
+        lon_min, lon_max = lon_max, lon_min
+
+    poly = box(
+        lon_min - lon_margin, max([lat_min - lat_margin, -90]), lon_max + lon_margin, min([lat_max + lat_margin, 90])
+    )
+    return poly
 
 
 def check_antimeridean(poly: Polygon) -> list[Polygon]:
+    """Check if the provided polygon crosses the antimeridian and split it if it does."""
```

Comment:
function name could be improved, e.g. `split_for_antimeridian`; "check" makes me expect that the function returns a boolean